src/System.Private.CoreLib/shared/System/IO/FileNotFoundException.cs

```diff
@@ -48,6 +50,7 @@ public override string Message
             get
             {
                 SetMessageField();
+                Debug.Assert(_message != null, "_message was null after calling SetMessageField");
```

Comment:
Why not instead make `SetMessageField()` return the message, then you don't need to add an assert.

I would just inline it

Reply:
> I would just inline it

+1

Reply:
Actually, it's also called from here:
https://github.com/dotnet/coreclr/blob/69649ed47f63b76c725b487ea21ea74713388d3b/src/System.Private.CoreLib/src/System/IO/FileNotFoundException.CoreCLR.cs#L17
so nevermind on manually inlining it.  What you have seems fine.

Reply:
FWIW, I do see why it is necessary to call `SetMessageField` in the `.CoreCLR.cs` constructor. It should be fine to create the message lazily here.

Reply:
That makes sense, too.

Reply:
So from what I understand, we should just leave it as is?

Reply:
Leaving as is.